sdlib/d/gc/bin.d

```diff
@@ -33,17 +33,21 @@ struct Bin {
 		mutex.lock();
 		scope(exit) mutex.unlock();
 
-		auto slab = (cast(Bin*) &this).getSlab(arena, emap, sizeClass);
-		if (slab is null) {
-			return null;
-		}
+		auto remainBuffer = buffer;
```

Comment:
If you need to create a local variable to track thing, it'd be simple for it to be the number of items allocated.

Reply:
You'd avoid a cast and various computations.